src/js/components/navbars.js

```diff
@@ -85,48 +86,127 @@
-   *  <div class="navbar navbar-app navbar-absolute-bottom">
-   *    <!-- ... -->
-   *  </div>
-   * ```
-   */
+  * @directive navController
+  * @restrict C
+  * @description
+  *
+  * Directive to bind controller to navbar
+  *
+  * ``` html
+  *  <div class='navbar navbar-app navbar-absolute-top' nav-controller>
+  *    <!-- ... -->
+  *  </div>
+  * ```
+  */
+  module.directive('navController', function() {
+    return {
+      restrict: 'A',
+      controller: function() {
+        this.registerController = function(name, ctrl) {
+          this[name] = ctrl;
+        };
+      },
+      link: function(scope) {}
+    };
+  });
+
+ /**
+  * @directive navbarAbsoluteTop
+  * @restrict C
+  * @description
+  *
+  * Setup absolute positioned top navbar.
+  *
+  * ``` html
+  *  <div class='navbar navbar-app navbar-absolute-top'>
+  *    <!-- ... -->
+  *  </div>
+  * ```
+  */
+
+ /**
+  * @directive navbarAbsoluteBottom
+  * @restrict C
+  * @description
+  *
+  * Setup absolute positioned bottom navbar.
+  *
+  * ``` html
+  *  <div class='navbar navbar-app navbar-absolute-bottom'>
+  *    <!-- ... -->
+  *  </div>
+  * ```
+  */
   angular.forEach(['top', 'bottom'], function(side) {
     var directiveName = 'navbarAbsolute' + side.charAt(0).toUpperCase() + side.slice(1);
     module.directive(directiveName, [
       '$rootElement',
       function($rootElement) {
         return {
           restrict: 'C',
-          link: function(scope) {
+          require: 'navController',
+          link: function(scope, elem, attr, ctrl) {
             $rootElement.addClass('has-navbar-' + side);
             scope.$on('$destroy', function() {
               $rootElement.removeClass('has-navbar-' + side);
             });
+            if (!global) global = ctrl;
+            global.registerController(side, new (function() {
+              this.hide = function() {
+                $rootElement.removeClass('has-navbar-' + side);
+                elem.addClass('ng-hide');
+              };
+              this.show = function() {
+                elem.removeClass('ng-hide');
+                $rootElement.addClass('has-navbar-' + side);
+              };
+              this.slideout = function() {
+                $rootElement.removeClass('has-navbar-' + side);
+                elem.addClass('ng-hide');
+              };
+              this.slidein = function() {
+                elem.removeClass('ng-hide');
+                $rootElement.addClass('has-navbar-' + side);
+              };
+            })());
           }
         };
       }
     ]);
   });
+
+  /**
+   * @directive [hide|show|slidein|slideout][Top|Bottom|Both][Nav]
+   * @restrict C
+   * @description
+   *
+   * Toggle each(or both) nabar(s) with or without animations
+   *
+   * ``` html
+   *  <view class='[hide|show|slidein|slideout]-[top|bottom|both]-[nav]'>
+   *    <!-- ... -->
+   *  </view>
+   * ```
+   */
+  angular.forEach(['Top', 'Bottom', 'Both'], function(nav) {
```

Comment:
Just remembered how I used to do it. So applying the override outside of the ```ng-view``` required extra global scope and/or ```ui-state``` tracking. So I had to use a combination of kludgy ```ng-if```'s and ```ui-state``` changes to get it to work. This is not only less than ideal but it is severely limited in the scope of its usefulness.